lib/open_food_network/scope_variants_for_search.rb

```diff
@@ -68,11 +70,27 @@ def scope_to_eligible_for_subscriptions_in_distributor
       scope_variants_to_distributor(@variants, distributor)
     end
 
+    def scope_to_in_stock_only
+      @variants = @variants.where(variant_in_stock_sql)
+    end
+
     def scope_variants_to_distributor(variants, distributor)
       scoper = OpenFoodNetwork::ScopeVariantToHub.new(distributor)
       # Perform scoping after all filtering is done.
       # Filtering could be a problem on scoped variants.
       variants.each { |v| scoper.scope(v) }
     end
+
+    def variant_in_stock_sql
+      "EXISTS (
+        SELECT 1
+        FROM variant_overrides
+        WHERE variant_id = spree_variants.id AND (on_demand = true OR count_on_hand > 0)
```

Comment:
I'm not sure about this part. One variant can be sold through multiple shops and it can have multiple variant overrides, one for each shop. But here we are just checking if any shop has stock, even if the shop we are creating the order for doesn't have stock, right?

Reply:
Thanks @mkllnk I'm not very familiar with this so I'm not sure. Would something like below work? If so I could add more test coverage.

```
def variant_in_stock_sql
  "EXISTS (#{variant_override_in_stock_sql}) OR EXISTS (
    SELECT 1
    FROM spree_stock_items
    WHERE variant_id = spree_variants.id AND (backorderable = true OR count_on_hand > 0)
  )"
end

def variant_override_in_stock_sql
  sql = "SELECT 1
         FROM variant_overrides
         WHERE variant_id = spree_variants.id AND (on_demand = true OR count_on_hand > 0)" 
  sql += " AND hub_id = #{distributor.id}" if distributor.present?
  sql
end
```

Does `spree_stock_items` need to take into account whether the `distributor` is present or not as well?

Reply:
That looks good to me. The stock items don't the the distributor. That's the stock defined by the producer.